src/Orleans.Runtime/Placement/DeploymentLoadPublisher.cs

```diff
@@ -224,8 +231,33 @@ private void OnSiloStatusChange(SiloAddress updatedSilo, SiloStatus status)
                 this.publishTimer.Dispose();
 
             SiloRuntimeStatistics ignore;
-            periodicStats.TryRemove(updatedSilo, out ignore);
+            PeriodicStatistics.TryRemove(updatedSilo, out ignore);
             NotifyAllStatisticsChangeEventsSubscribers(updatedSilo, null);
         }
+
+        void ILifecycleParticipant<ISiloLifecycle>.Participate(ISiloLifecycle lifecycle)
+        {
+            var tasks = new List<Task>();
+            lifecycle.Subscribe(nameof(DeploymentLoadPublisher), ServiceLifecycleStage.Active, OnActiveStart, OnActiveStop);
+
+            Task OnActiveStart(CancellationToken ct)
+            {
+                tasks.Add(this.ScheduleTask(this.PublishStatistics));
+                this.siloStatusOracle.SubscribeToSiloStatusEvents(this);
+                return Task.CompletedTask;
+            }
+
+            async Task OnActiveStop(CancellationToken ct)
+            {
+                this.publishTimer.Dispose();
```

Comment:
Suggest calling this.Dispose() to avoid duplicate code and setting publishTimer = null to avoid double dispose.

Reply:
It's a readonly, so how about I add a `IAsyncTimer.Cancel()` method and call that here instead? I think that has a clearer intention.

Double-dispose is 'guaranteed' (by [IDisposable docs](https://docs.microsoft.com/en-us/dotnet/api/system.idisposable.dispose?view=netframework-4.8)) to be permissible:

> If an object's Dispose method is called more than once, the object must ignore all calls after the first one. The object must not throw an exception if its Dispose method is called multiple times. Instance methods other than Dispose can throw an ObjectDisposedException when resources are already disposed.

Reply:
Agreed.  Just need to ensure our impl follows that rule.